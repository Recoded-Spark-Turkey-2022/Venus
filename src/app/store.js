// eslint-disable-next-line no-unused-vars
import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit';
import listingReducer from '../features/listings/listingSlice';

export const store = configureStore({
  reducer: {
    listing: listingReducer,
  },
  // eslint-disable-next-line no-shadow
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: false,
    }),
});
