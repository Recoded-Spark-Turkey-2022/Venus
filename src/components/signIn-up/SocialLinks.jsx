import React from 'react';
import { useNavigate } from 'react-router-dom';
import { useSelector } from 'react-redux';
import { signInWithPopup } from 'firebase/auth';
import { doc, getDoc, serverTimestamp, setDoc } from 'firebase/firestore';
import {
  authentication,
  db,
  faceBookProvider,
  googleProvider,
} from '../../firebase/firebase.config';

import './Signup-in.css';
import {
  selectUserLoggedIn,
  selectUserName,
} from '../../features/userSlice/userSlice';

const SocialLinks = () => {
  const navigate = useNavigate();
  const userloggedIn = useSelector(selectUserLoggedIn);
  const userName = useSelector(selectUserName);
  console.log(userloggedIn, userName);

  const onGoogleAuth = async () => {
    try {
      const result = await signInWithPopup(authentication, googleProvider);
      const { user } = result;
      // if (user.uid) {

      // }

      const docRef = doc(db, 'users', user.uid);
      const docSnap = await getDoc(docRef);

      if (!docSnap.exists()) {
        await setDoc(docRef, {
          name: user.displayName,
          email: user.email,
          timeStamp: serverTimestamp(),
        });
      } else {
        console.log('you already signed in');
      }
      navigate('/', { replace: true });
    } catch (error) {
      console.log(error);
    }
  };

  const onFacebookAuth = async () => {
    try {
      const result = await signInWithPopup(authentication, faceBookProvider);
      const { user } = result;
      /* Checking if we have a reference to that user */
      const docRef = doc(db, 'users', user.uid);
      const docSnap = await getDoc(docRef);
      if (!docSnap.exists()) {
        await setDoc(docRef, {
          name: user.displayName,
          email: user.email,
          timeStamp: serverTimestamp(),
        });
      } else {
        console.log('you already signed in');
      }
      navigate('/', { replace: true });
      console.log(user);
    } catch (error) {
      console.log(error.message);
    }
  };
  return (
    <div className="ContainerBtn">
      <button className="GoogleBtn" onClick={onGoogleAuth} type="button">
        <ion-icon size="large" name="logo-google" />
      </button>
      <h1 className="or">OR</h1>
      <button className="FacebookBtn" onClick={onFacebookAuth} type="button">
        <ion-icon size="large" className="" name="logo-facebook" />
      </button>
    </div>
  );
};

export default SocialLinks;
