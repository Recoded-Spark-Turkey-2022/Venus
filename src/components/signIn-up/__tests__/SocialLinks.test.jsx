import React from 'react';
import renderer from "react-test-renderer";
import { Provider } from 'react-redux';
import { store } from '../../../app/store';
import { BrowserRouter as Router } from "react-router-dom";
import SocialLinks from "../SocialLinks";



  it(`renders correctly` , () => {
    const tree = renderer.create(
    <Router>
      <Provider store={store}>
        <SocialLinks />
        </Provider>
    </Router>);
    expect(tree).toMatchSnapshot();
});