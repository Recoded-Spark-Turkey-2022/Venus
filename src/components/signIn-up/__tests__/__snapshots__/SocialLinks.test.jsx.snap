// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="ContainerBtn"
>
  <button
    className="GoogleBtn"
    onClick={[Function]}
    type="button"
  >
    <ion-icon
      name="logo-google"
      size="large"
    />
  </button>
  <h1
    className="or"
  >
    OR
  </h1>
  <button
    className="FacebookBtn"
    onClick={[Function]}
    type="button"
  >
    <ion-icon
      className=""
      name="logo-facebook"
      size="large"
    />
  </button>
</div>
`;
